### PR TITLE
Convert InvalidOperationException to InvalidDataException for form reader

### PR DESCRIPTION
**Pull request by Marusyk**
For now, I wrapped the `InvalidOperationException` in an `InvalidDataException` in 2 places for review.
Please clarify which one should be removed, from `GetDecodedString` or `ReadFormAsync`? 

One more possible solution is adding try/catch to the `ParseValuesSlow` method. 👎 

Addresses #19962

Please review,
Thank you in advance

### CODE REVIEW

File: src/Http/WebUtilities/src/FormPipeReader.cs [hunk 1]
```diff
@@ -97,11 +97,18 @@ public FormPipeReader(PipeReader pipeReader, Encoding encoding)
                     {
                         ParseFormValues(ref buffer, ref accumulator, readResult.IsCompleted);
                     }
-                    catch
+                    catch (InvalidOperationException ex)
```
Comment (Tratcher) [inline src/Http/WebUtilities/src/FormPipeReader.cs]: Revert this one, it's better to handle it in GetDecodedString.

File: src/Http/WebUtilities/src/FormPipeReader.cs [hunk 2]
```diff
@@ -377,10 +384,17 @@ private string GetDecodedString(ReadOnlySpan<byte> readOnlySpan)
...
+                {
+                    var bytes = UrlDecoder.DecodeInPlace(span, isFormEncoding: true);
+                    span = span.Slice(0, bytes);
 
-                return _encoding.GetString(span);
+                    return _encoding.GetString(span);
+                }
+                catch (InvalidOperationException ex)
+                {
+                    throw new InvalidDataException("The path contains null characters.", ex);
```
Comment (Tratcher) [inline src/Http/WebUtilities/src/FormPipeReader.cs]:
```suggestion
                    throw new InvalidDataException("The form value contains invalid characters.", ex);
```

**Review by Tratcher — COMMENTED** [review]
Test?

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks

**Comment by abatishchev** [comment]
hi @Tratcher, do you know in which version of ASP.NET MVC this change will be available? Only in 5 or will be backported to 3?

**Pull request merged**